File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/DungeonPartyFinderListenRequestMessage.scala





















// Generated on 05/03/2013 10:34:46
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait DungeonPartyFinderListenRequestMessage extends MessageDeserializer {
  val messageId = 6246

  protected trait Message extends BaseMessage with Serializable {
    val messageId = DungeonPartyFinderListenRequestMessage.messageId

    val dungeonId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(dungeonId)
    }

  }

  protected def create(dungeonId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val dungeonId = buf.readShort()

    create(dungeonId)
  }

}

object DungeonPartyFinderListenRequestMessage extends DungeonPartyFinderListenRequestMessage {
  case class MessageImpl(dungeonId: Short) extends Message
  type Target = MessageImpl

  override def create(dungeonId: Short) = MessageImpl(dungeonId)

  def apply(dungeonId: Short) = create(dungeonId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/DungeonPartyFinderListenRequestMessage.java


// Generated on 05/08/2013 19:37:48
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class DungeonPartyFinderListenRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6246;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short dungeonId;
    
    public DungeonPartyFinderListenRequestMessage() { }
    
    public DungeonPartyFinderListenRequestMessage(short dungeonId) {
        this.dungeonId = dungeonId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(dungeonId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        dungeonId = buf.readShort();
        if (dungeonId < 0)
            throw new RuntimeException("Forbidden value on dungeonId = " + dungeonId + ", it doesn't respect the following condition : dungeonId < 0");
    }
    
}
